internal/generator/fri/template/fri.go.tmpl

```diff
@@ -0,0 +1,700 @@
+import (
+	"errors"
+	"fmt"
+	"hash"
+	"math/big"
+	"math/bits"
+
+	"github.com/consensys/gnark-crypto/accumulator/merkletree"
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark-crypto/ecc/{{.Name}}/fr"
+	"github.com/consensys/gnark-crypto/ecc/{{.Name}}/fr/fft"
+	fiatshamir "github.com/consensys/gnark-crypto/fiat-shamir"
+)
+
+var (
+	ErrLowDegree            = errors.New("the fully folded polynomial in not of degree 1")
+	ErrProximityTestFolding = errors.New("one round of interaction failed")
+	ErrOddSize              = errors.New("the size should be even")
+	ErrMerkleRoot           = errors.New("merkle roots of the opening and the proof of proximity don't coincide")
+	ErrMerklePath           = errors.New("merkle path proof is wrong")
+	ErrRangePosition        = errors.New("the asked opening position is out of range")
+)
+
+const rho = 8
+
+var NbRounds = 1
```

Comment:
is it a constant? if it's not, should be made a parameter to the functions where it's needed.